### PR TITLE
Fix makeIntIndex, benchmark get loc

### PR DESCRIPTION
**Pull request by toobaz**
- [x] tests passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

``makeIntIndex`` currently produces a monotonic index, and hence tests common to different classes actually test very different things (i.e. the sorted vs. unsorted code paths). I had to fix this (and to fix the tests which erroneously relied on the specific content of ``makeIntIndex(k)``) in order to add meaningful performance tests for ``get_loc``.

### CODE REVIEW

File: pandas/tests/indexing/test_floats.py
Comment (jreback) [inline pandas/tests/indexing/test_floats.py]: rather than change this, wouldn't it be better to simply fix the ``make*`` ones? this is pretty much an anti-patter and the reason we have the helpers.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_floats.py
Comment (toobaz) [reply pandas/tests/indexing/test_floats.py]: The only way to "fix" ``makeIntIndex`` for this test to work is to leave it as it is, but as I stated, it limits comparability across index types, since others just return random stuff, i.e. unsorted.

This test does something different than using the ``makeIntIndex`` to get "some index filled with integers": it assumes that the result has a specific content. I see _this_ as an anti-pattern.

File: pandas/util/testing.py
Comment (jreback) [inline pandas/util/testing.py]: why is this needed?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_floats.py
Comment (jreback) [reply pandas/tests/indexing/test_floats.py]: why are you removing testing of RangeIndex? (yes you convert to it, but its not very explicit)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_floats.py
Comment (toobaz) [reply pandas/tests/indexing/test_floats.py]: It's just simpler, but OK, I can make it explicit

File: pandas/util/testing.py
Comment (toobaz) [reply pandas/util/testing.py]: ... because otherwise the index comes with an element anyway. But I'll reformulate

File: pandas/tests/indexing/test_floats.py
Comment (toobaz) [reply pandas/tests/indexing/test_floats.py]: (done), ping

File: pandas/util/testing.py
Comment (jreback) [inline pandas/util/testing.py]: why are you changing this? sure this is an int index, but the ordering is not expected. Revert this, and if you really need it then make a fixture in the appropriate places

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/util/testing.py
Comment (toobaz) [reply pandas/util/testing.py]: > the ordering is not expected.

Why do _you_ expect an index of int (and not of float, str) to be sorted?

What don't you understand in the [description](https://github.com/pandas-dev/pandas/pull/19483#issue-293353518) of this PR?

Sure, I can write all the fixtures you like. It just doesn't make any sense.

Comment (jreback) [reply pandas/util/testing.py]: because for testing using something like arange is the expected input.

Comment (jreback) [reply pandas/util/testing.py]: this has nothing to do with the description. you are free to use sorted, unique, not-sorted whatever in a test, but that should be explicit. you are changing a global default.

Comment (toobaz) [reply pandas/util/testing.py]: > because for testing using something like arange is the expected input.

If _you_ expect that, fine to me: shall I then change ``makeFloatIndex`` and ``makeStringIndex`` to return sorted indexes, so that when I shuffle them it is "explicit" that they are unsorted? I couldn't care less if standard test indexes are sorted or not, as long as they all are and I don't need to add useless lines of codes in tests.

Currently not only it's not "explicit" if tests run on sorted indexes or not - it actually depends on the type.

Comment (toobaz) [reply pandas/util/testing.py]: In general when you run a test on different types you want to change _only_ the type. Not other characteristics of the index. Then sure, there are some unavoidable differences (e.g. a ``RangeIndex`` _is_ sorted)... but apart from that, they should be as comparable as possible, so that tests can really test the different code paths.

Comment (toobaz) [reply pandas/util/testing.py]: > because for testing using something like arange is the expected input.

Actually, the concept itself of "expected input" (referring to _content_) is completely wrong in a test.

Comment (jreback) [reply pandas/util/testing.py]: pls do it my way. you are changing a function that is public facing for no real reason.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/util/testing.py
Comment (toobaz) [reply pandas/util/testing.py]: > pls do it my way. you are changing a function that is public facing for no real reason.

Oh come on Jeff, the function is clearly not part of the API. I gave "real reasons", you didn't. I even proposed an [alternative](https://github.com/pandas-dev/pandas/pull/19483#discussion_r165623414), which you didn't even comment... you're certainly not going to convince me by asking "pls do it my way".

By the way, you are a core dev, you can always merge and then change (e.g. the new asv tests), so if you don't like my PR _and_ don't like to discuss... you don't actually need to waste your time here, just merge and fix to your liking.

**Comment by toobaz** [comment]
@jreback You realize that the asv test is broken in that it never tests unsorted ``Int64Index``, yeah?

**Comment by jreback** [comment]
and that change should be in the asv itself

**Comment by toobaz** [comment]
fine

**Comment by jorisvandenbossche** [comment]
@jreback if you make changes to a PR before merging (which is OK, I also do that from time to time), please push to this branch on github and merge here, otherwise it is really un-transparent to see what has been merged (the merge script has even an option to do this).

**Pull request closed without merging**
